addons/purchase_stock/models/stock.py

```diff
@@ -42,7 +42,7 @@ def _get_price_unit(self):
             if line.taxes_id:
                 price_unit = line.taxes_id.with_context(round=False).compute_all(price_unit, currency=line.order_id.currency_id, quantity=1.0)['total_void']
             if line.product_uom.id != line.product_id.uom_id.id:
-                price_unit *= line.product_uom.factor / line.product_id.uom_id.factor
+                price_unit = line.product_uom._compute_price(line.price_unit, line.product_id.uom_id)
```

Comment:
Should be `price_unit` and not `line.price_unit` to consider taxes.

Reply:
yes you are right, will try to modify it as I made the change from github interface.